### PR TITLE
Add create endpoint

### PR DESCRIPTION
**Pull request by nspektor**
Here is the basic version of the create endpoint, I have not added testing when a promotion is related to a product because I was not sure how we would want to do that, @liondon any insight? What would a JSON body look like for a promotion associated with just 1 product? with multiple products?

### CODE REVIEW

**Comment by VickySteeves** [comment]
So to test that this works locally, I created the file `01-halloween.json` with the following body:

```
{
    "id": 2,
    "title": "Halloween Special",
    "description": "Some items off in honor of the spookiest month.",
    "promo_code": "hween",
    "promo_type": "DISCOUNT",
    "amount": 25,
    "start_date": "2020-10-20",
    "end_date": "2020-11-01",
    "is_site_wide": true
}
```

I then used CURL to test POSTing this to our service:  

```
curl -v http://127.0.0.1:5000/promotions -d @01-halloween.json -H 'Content-Type: application/json'
```

And I got this back:
```
*   Trying 127.0.0.1...
* TCP_NODELAY set
* Connected to 127.0.0.1 (127.0.0.1) port 5000 (#0)
> POST /promotions HTTP/1.1
> Host: 127.0.0.1:5000
> User-Agent: curl/7.58.0
> Accept: */*
> Content-Type: application/json
> Content-Length: 272
> 
* upload completely sent off: 272 out of 272 bytes
* HTTP 1.0, assume close after body
< HTTP/1.0 201 CREATED
< Content-Type: application/json
< Content-Length: 266
< Server: Werkzeug/1.0.1 Python/3.6.9
< Date: Mon, 19 Oct 2020 16:06:25 GMT
< 
{"amount":25,"description":"Some items off in honor of the spookiest month.","end_date":"Sun, 01 Nov 2020 00:00:00 GMT","id":1,"is_site_wide":true,"promo_code":"hween","promo_type":"DISCOUNT","start_date":"Tue, 20 Oct 2020 00:00:00 GMT","title":"Halloween Special"}
* Closing connection 0
```

So looks good to me! Does what it says it does.

**Comment by VickySteeves** [comment]
The only problem is when I ran the tests, it only covered 68% because of a missing dependency:

```
   ModuleNotFoundError: No module named 'factory'

Name                  Stmts   Miss  Cover   Missing
---------------------------------------------------
service/__init__.py      16      3    81%   27-30
service/models.py        59     16    73%   83, 96, 115-130
service/service.py       26     13    50%   29, 47-56, 72-75
---------------------------------------------------
TOTAL                   101     32    68%
-----------------------------------------------------------------------------
4 tests run in 0.609 seconds. 
1 error (3 tests passed)
```

Would this stop merging? I know we need coverage above, what, 98% at the end of sprint 1?

File: requirements.txt [hunk 1]
```diff
@@ -9,5 +9,6 @@ psycopg2-binary==2.8.3
 nose==1.3.7
 rednose==1.3.0
 pinocchio==0.4.2
+factory-boy==2.12.0
```
Comment (nspektor) [inline requirements.txt:12]: dependency is here

**Comment by liondon** [comment]
@nspektor Hey, regarding how to test the relationship, I put my thoughts at our slack channel:
https://nyu-devops-fall20.slack.com/archives/C01B93WGKK8/p1602978758025200

**Pull request merged**